extensions/amp-analytics/0.1/visibility-manager.js

```diff
@@ -408,6 +424,18 @@ export class VisibilityManagerForDoc extends VisibilityManager {
   }
 
   /** @override */
+  getRootLayoutBox() {
+    if (getMode(this.ampdoc.win).runtime == 'inabox') {
+      const root = this.ampdoc.getRootNode();
+      const rootElement = dev().assertElement(
+          root.documentElement || root.body || root);
+      return this.viewport_.getLayoutRect(rootElement);
+    }
+    const size = this.viewport_.getSize();
+    return layoutRectLtwh(0, 0, size.width, size.height);
```

Comment:
That's where a lot of confusion rests. In theory, `<html>` is scrollable and thus equal to viewport, while `<body>` spans the whole doc. But, of course, your interpretation is also logically plausible. I don't mind "in-a-box" for now, because we'll likely have to revisit viewport implementation there as we discussed, so I appreciate these markers that need to be considered while doing that work. That being said, I'll be happy to change the meaning of `elementHeight` - that's the only real metric affected here.